src/index.ts

```diff
@@ -727,43 +756,59 @@ const buildOption = (
 	return `${option}=${compiledValue}`;
 };
 
-const handleExpandOptions = (
-	expand: ODataOptions,
+const handleOptions = (
+	optionOperation: keyof typeof ODataOptionCodeExampleMap,
+	options: ODataOptions,
 	parentKey: string,
 ): string => {
-	if (parentKey.endsWith('/$count')) {
-		deprecated.countInExpand();
-	}
-	if (expand.hasOwnProperty('$count')) {
-		const keys = Object.keys(expand);
+	if (options.hasOwnProperty('$count')) {
+		const keys = Object.keys(options);
 		if (keys.length > 1) {
 			throw new Error(
-				`When using '$expand: a: $count: ...' you can only specify $count, got: '${JSON.stringify(
-					keys,
-				)}'`,
+				`When using '${
+					ODataOptionCodeExampleMap[optionOperation]
+				}' you can only specify $count, got: '${JSON.stringify(keys)}'`,
 			);
 		}
-		expand = expand.$count!;
+
+		options = options.$count!;
 		parentKey += '/$count';
+		if (
+			// TODO: Remove in the next major and so that it throws for $expands as well.
```

Comment:
Why should it throw for `$expand`s? Is this related to the `deprecated.countInExpand();`? If so then it'd be worth mentioning that here imo

Reply:
No it's not related to the `deprecated.countInExpand` @Page- .
It's that $select/$top doesn't make sense inside $count, but before this PR, users could do:
```
testExpand(
	a: $count:
		$select: 'a'
		$filter: b: 'c'
	"a/$count($select=a;$filter=b eq 'c')"
)
```
and even thugh pine atm doesn't error and just ignores it, imo it doesn't make sense to allow it.
On the other hand pine atm errors if you try to do the same as part of a `$orderby`.

The only reason I left it as a TODO, was to not break unexpectedly clients that might be doing this already.

Depending on how preservative or strict we want to be, we could range from completely removing this `if` and allow users to pass anything, to start throwing right away for $expands & $orderby with this minor PR.

Reply:
If we plan to remove it in the next major then can you please add a deprecated message so we know what to replace with a `throw new Error(...)` on the major